### PR TITLE
Add docs on pylint

### PR DESCRIPTION
**Pull request by vladsaveliev**
Will update if we adopt another linter or code style (after some feedback from Hail), or if we automate style formatting.

### CODE REVIEW

File: python_code_style.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+2. Install a PyCharm "Pylint" plugin ([Preferences] > [Plugins] > search for "Pylint"). It will automatically find the pylint executable. A new tool window called Pylint will be added. You can use it to scan the project:
+   
+<img src="assets/pycharm_pylint_tool_window.png" width="700"/>
+
+3. You can also enable real-time inspections by going to:
+   Go to Preferences > Editor > Inspections > enable "Pylint real-time scan". However, it's not recommended as it's has a negative impact on system performance.
+
+4. Pylint has a pre-commit hook integrated into a Pycharm commit modal window (cmd+K). Make sure the "Scan with Pylint" checkbox is enabled:
+   
+<img src="assets/pycharm_pylint_pre_commit.png" width="700"/>
```
[the remark this answers is not in the document]
Comment (vladsaveliev) [reply python_code_style.md:102]: Sorry bot, but GitHub markdown doesn't like the image width specifiers in the markdown img elements, so gotta go with HTML.

File: python_code_style.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+# Python Code Style
+
+We use [pylint](https://www.pylint.org/) to perform automatic code checks on our
+repositories. It helps enforce code standard, and also does some static code
+analysis for potential programming errors.
```
Comment (lgruen) [inline python_code_style.md:5]:
```suggestion
repositories. It helps us in implementing a consistent coding style throughout
our code base, and also does some static code analysis to catch potential
programming errors.
```

File: python_code_style.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+## Setting up a new project
+
+When creating a new repository that contains Python code, please add into the
+repository root folder a file named `.pylintrc` with the following contents:
```
Comment (lgruen) [inline python_code_style.md:10]: How about just putting a `.pylintrc` file into this repository, that people can easily copy? Maybe just provide the command to copy the file from the GitHub blob.

File: python_code_style.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+You may want to disable additional inspections if pylint hits a false positive.
+For example, it might fail to recognise imports of third-party libraries, in
+which case can add E0401 ("Unable to import") to the comma-separated list:
+
+```
+disable=f-string-without-interpolation,inherit-non-class,too-few-public-methods,E0401
+```
+
+In addition to `.pylintrc`, please create a GitHub Actions CI workflow under
+`.github/workflows/main.yaml` with the following contents (or just append the
```
Comment (lgruen) [inline python_code_style.md:49]: Same here; it's slightly easier to copy a file than copy & pasting content :).

File: python_code_style.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+checks in the web interface.
+
+<img src="assets/github_pylint_check.png" width="400"/>
+
+## Running Pylint
+
+To install Pylint into your environment, run:
+
+```
+pip install pylint pylint-quotes
```
Comment (lgruen) [inline python_code_style.md:81]: What does `pylint-quotes` do?

**Review by lgruen — APPROVED** [review]
This looks great, just added a few minor comments (therefore not merging now, despite already approving).

File: python_code_style.md (hunk 5 above)
Comment (vladsaveliev) [reply python_code_style.md:82]: Checks the consistency of quotes (single or double). I set up to single ones. Let me know if that's unnecessary!

File: python_code_style.md (hunk 4 above)
Comment (vladsaveliev) [reply python_code_style.md:50]: Didn't want to add it as a file it to avoid confusion with _real_ GitHub workflow yaml for this repository :)

Comment (vladsaveliev) [reply python_code_style.md:50]: Could add it into a root and provide commands to create the directory structure and to wget the file from the blob as you suggested.

Comment (vladsaveliev) [reply python_code_style.md]: Done!

File: python_code_style.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
+# Python Code Style
+
+We use [pylint](https://www.pylint.org/) to perform automatic code checks on our
+repositories. It helps us in implementing a consistent coding style throughout
+our code base, and also does some static code analysis to catch potential
+programming errors.
+
+## Setting up a new project
+
+When creating a new repository that contains Python code, please add this
```
[the remark this answers is not in the document]
Comment (vladsaveliev) [reply python_code_style.md:10]: @pdiakumis, what do you think about disabling this one too?

**Pull request merged**